actions/reportportal_submit_execution_results/action.yml

```diff
@@ -0,0 +1,126 @@
+          echo "The DROUTE_PASSWORD secret is not defined for this repository"
+          exit 1
+        fi
+
+        ## construct name
+        REPORTPORTAL_RUN_NAME="${{ inputs.run_name }}"
+        if [[ -z "${{ inputs.run_name }}" ]]; then
+            BRANCH_NAME="$GITHUB_REF_NAME"
+            TMP="$ACTOR-$BRANCH_NAME"
+            REPORTPORTAL_RUN_NAME=$(echo $TMP | awk '{print tolower($0)}')
+        fi
+        echo "test run name: $REPORTPORTAL_RUN_NAME"
+
+        # pull JUnitXML result files from gcp bucket
+        gcloud storage cp "${{ secrets.GCP_BUILD_ASSETS }}/${{ inputs.run_id }}/test-results" .
+
+        # verify results folder contains XML result files
+        readarray -d '' RESULTS < <(find "test-results" -type f -name "*.xml" -print0)
+        if [[ ${#RESULTS[@]} == 0 ]]; then
+          echo "Results folder 'test-results' does not contain any XML result files:"
+          ls -A "test-results"
+          echo "::warning title=$GITHUB_JOB - MISSING RESULT FILES::Results folder did not contain any XML result files"
+          exit 1
+        fi
+
+        # compose our metadata file
+        jq -n '{
+          "targets": {
+            "reportportal": {
+              "disabled": false,
+              "config": {
+                "hostname": "$REPORTPORTAL_HOST",
+                "project": "${{ inputs.project }}"
+              },
+              "processing": {
+                "apply_tfa": true,
+                "merge_launches": true,
+                "simple_xml": false,
+                "auto_dashboard": false,
+                "launch_with_class_name": true,
+                "property_filter": [".*"],
+                "debug": true,
+                "launch": {
+                  "name": "${{ inputs.run_name }}",
+                  "description": "GitHub run url: ${{ inputs.run_url}}",
+                  "attributes": [
+                    {"key": "Product Version", "value": "${{ inputs.version }}"},
+                  ]
+                }
+              }
+            }
+          }
+        }' > datarouter.json
+
+        # capture output and verify success
+        SUCCESS=0
+        LAUNCH_URL = $(droute send --metadata datarouter.json \
+                                   --results 'test-results/*.xml' \
+                                   --username "$DROUTE_USERNAME" \
+                                   --password "$DROUTE_PASSWORD" \
+                                   --url "$DROUTE_URL" \
+                                   --verbose \
+                                   --wait | yq -r '.targets[] | select(.name == "reportportal") | .events[] | select(.component == "reportportal-connector") | .message' | jq -r '.[0]."launch_url"')
+        echo "launch_url=$LAUNCH_URL" >> "$GITHUB_OUTPUT"
+
+        -- step-status.sh "$STEP_STATUS" || SUCCESS="$?"
+        echo "status=$SUCCESS" >> "$GITHUB_OUTPUT"
+        exit "$SUCCESS"
+      env:
+        DROUTE_USERNAME: ${{ inputs.cred_username }}
```

Comment:
The DROUTE_* and REPORTPORTAL_HOST are all secrets in GHA. I think we need to pass them in as inputs for the action.

Reply:
I'm confused by your wording.  right now, the DROUTE_* envs _are_ being set by inputs to the action (e.g. `DROUTE_USERNAME: ${{ inputs.cred_username }}`
Are you suggesting that I should NOT use inputs to the action, and instead use the secrets directly?  e.g `DROUTE_USERNAME: ${{ secrets.DROUTE_USERNAME }}`

Reply:
so you're saying:
`REPORTPORTAL_HOST: ${{ inputs.reportportal_host }}`

and add the "reportportal_host" as an input?

Reply:
yes